src/Schema/Table.php

```diff
@@ -938,6 +939,14 @@ protected function _addForeignKeyConstraint(ForeignKeyConstraint $constraint): s
             );
 
         $name = $this->normalizeIdentifier($name);
+        if (array_key_exists($name, $this->_fkConstraints)) {
```

Comment:
I'd replace `array_key_exists()` with `isset()` since it's historically preferable and we don't expect null values.